docs/reference.rst

```diff
@@ -923,11 +923,34 @@ The following fields are supported for image sections:
 ``filename`` (required)
   Name of the image file (relative to bundle content).
 
+  .. note::
+    While RAUC can automatically detect the image type from filename extensions
+    for backward compatibility, it is recommended to explicitly specify the
+    ``type`` field for new projects or in cases where all target systems
+    are running a new enough version.
+
+  .. note::
+    This field is not required and not supported for images with ``type=emptyfs``,
+    which creates an empty filesystems without source data.
+
   .. important::
-    RAUC uses the file name extension and the slot type to decide how to
+    If RAUC uses the file name extension and the slot type to decide how to
```

Comment:
What is the 'If' for here?

Reply:
I've created https://github.com/rauc/rauc/pull/1812 for this.